worker/src/instagram.rs

```diff
@@ -0,0 +1,81 @@
+use crate::utils::extract_signature;
+use anyhow::{anyhow, Result};
+use reqwest::header::{HeaderMap, AUTHORIZATION};
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct User {
+    pub username: String,
+    pub id: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Post {
+    pub sig: String,
+    pub sig_target: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CaptionWrapper {
+    pub caption: String,
+    pub permalink: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PostWrapper {
+    pub data: Vec<PostId>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PostId {
+    pub id: String,
+}
+
+pub async fn retrieve_post(handle: String, token: String) -> Result<(String, String, String)> {
+    let url = format!(
+        "https://graph.instagram.com/me?fields=id,username&access_token={}",
+        token
+    );
+
+    let client = reqwest::Client::new();
+    let res: User = client.get(Url::parse(&url)?).send().await?.json().await?;
+
+    if handle != res.username {
+        return Err(anyhow!(format!(
+            "Token is not authorized for handle: {}",
```

Comment:
I would the message to something like "mismatch between post author and caller", as this error message will be shown to the end user.

Reply:
Will change to a friendlier error message.